index.js

```diff
@@ -2,57 +2,78 @@
 
 module.exports = CSSselect;
 
-var Pseudos       = require("./lib/pseudos.js"),
-    DomUtils      = require("domutils"),
-    findOne       = DomUtils.findOne,
-    findAll       = DomUtils.findAll,
-    getChildren   = DomUtils.getChildren,
-    getSiblings   = DomUtils.getSiblings,
-    removeSubsets = DomUtils.removeSubsets,
-    falseFunc     = require("boolbase").falseFunc,
-    compile       = require("./lib/compile.js"),
-    compileUnsafe = compile.compileUnsafe,
-    compileToken  = compile.compileToken;
+var DomUtils       = require("domutils"),
+    falseFunc      = require("boolbase").falseFunc,
+    compileFactory = require("./lib/compile.js"),
+    browserAdapter = require("./browser-adapter"),
+    defaultCompile = compileFactory(DomUtils);
+
+function getAdapter(options){
+	if(options && options.adapter) return options.adapter;
+
+	return DomUtils;
+}
 
 function getSelectorFunc(searchFunc){
 	return function select(query, elems, options){
+		var adapter = getAdapter(options),
+		    compile,
+		    compileUnsafe;
+
+		if(adapter === DomUtils){
+			compile = defaultCompile;
+		} else {
+			compile = compileFactory(adapter);
+		}
+
+		compileUnsafe = compile.compileUnsafe;
```

Comment:
Not sure I can follow why separate functions are required. I would have approached this by

1. Making sure `options` is always an object (`options = options || {}`)
2. And ensuring its `adapter` property is set (`options.adapter = options.adapter || DomUtils`)